Source/Common/CommonBase.cpp

```diff
@@ -98,15 +98,26 @@ void CommonBase::MPISpinner()
 
 int MSGBUFSIZ = 1024;                  // Individual incoming message buffer
 char * MSGBUF = new char[MSGBUFSIZ];   // Pull it off the heap
+
+const int MAX_PROBE_IDLE_BACKOFF_US=100000; // Maximum is 100,000us, i.e. 10hz
+int probe_idle_backoff_us=0;
 for (;;) {
 // See if there are any MPI packets coming down the pipe
   MPI_Status status;                   // Note the multi-threaded MPI probe
   int flag;
   MPI_Iprobe(MPI_ANY_SOURCE,MPI_ANY_TAG,MPI_COMM_WORLD,&flag,&status);
   if (flag==0) {                       // Nothing there....
-    OnIdle();                          // Guess
+    if(HaveIdleWork()){
+      OnIdle();                          // Guess
+    }else{
+      probe_idle_backoff_us=std::min<int>(MAX_PROBE_IDLE_BACKOFF_US, std::max<int>(10, (probe_idle_backoff_us*5)/4));

```

Comment:
I would prefer to see some documentation about this, explaning how `probe_idle_backoff_us` decreases and increases incrementally as a function of workload.

Small note on styling - we generally try to adhere to the same style as neighbouring source when introducing modifications. In this case, lines of source in the Orchestrator source shouldn't span more than 79 characters, and we generally avoid underscores in variable names over camel case.

We do have a style guide, at https://github.com/POETSII/Orchestrator/wiki/Code-Style, but I don't want to dissuade you from contributing!

Reply:
I didn't really think about the schedule, it's just exponential backoff with a slow-ish
multiplier and a reset to zero on activity. I'm afraid there is no particular thought that went into it. So feel free to change it to better match work-load and so on.

In terms of style, I generally try to match where possible, but it is a balance
between fixing things for my ultimate goals (getting DPD working) versus trying
to package up into small patches that can be easily applied. So probably that
means more work to integrate the PRs, but I don't really expect them
to be directly applicable, as I don't really understand the orchestrator.
So feel free to improve however you want, and I'll just merge 1.0.0-alpha
back into my copy.